seacatauth/credentials/service.py

```diff
@@ -396,6 +396,27 @@ async def update_credentials(self, credentials_id: str, update_dict: dict, sessi
 				"message": "Data does not comply with update policy",
 			}
 
+		# Validate that at least phone or email will be specified after update
+		prev_cred_dict = await provider.get(credentials_id)
+		checking_dict = {
+			"prev_email": prev_cred_dict.get("email"),
+			"prev_phone": prev_cred_dict.get("phone"),
+			"new_email": update_dict.get("email"),
+			"new_phone": update_dict.get("phone")
+		}
+		email_specified = checking_dict["prev_email"] or checking_dict["new_email"]
```

Comment:
I think the correct logic should be:
```python
if "email" in update_dict and update_dict["email"] == "":
  email_specified = False
elif "email" not in current_dict or current_dict["email"] == "":
  email_specified = False
else:
  email_specified = True
```
and similarly for phone.

You probably do not need to create that extra `checking_dict`.